control/include/http/report_data.h

```diff
@@ -29,6 +29,9 @@ class ReportData {
  public:
   virtual ~ReportData() {}
 
+  // Get upstream tcp connection ip and port.
+  virtual bool GetDestinationIpPort(std::string* ip, int* port) const = 0;
```

Comment:
please comment on the format of ip in bytes

Reply:
Done.